compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -4174,37 +4174,76 @@ object Types {
 
     def tryCompiletimeConstantFold(using Context): Type = tycon match {
       case tycon: TypeRef if defn.isCompiletimeAppliedType(tycon.symbol) =>
-        def constValue(tp: Type): Option[Any] = tp.dealias match {
+        extension (tp : Type) def fixForEvaluation : Type =
+          tp.normalized.dealias match {
+            case tp : TermRef => tp.underlying
+            case tp => tp
+          }
```

Comment:
I suppose this is the bit that fixes constant termrefs types not being recognized?

Reply:
Yes. I'll add a comment there.